src/infra/core/src/repos/dataset_repository_local_fs.rs

```diff
@@ -82,11 +82,25 @@ impl DatasetRepositoryLocalFs {
         ))
     }
 
+    fn build_dataset(layout: DatasetLayout, event_bus: Arc<EventBus>) -> Arc<dyn Dataset> {
```

Comment:
Notice that I have severed ties between `DatasetRepository`s and `DatasetFactory`.

After making a bunch of changes to make `DatasetFactory` super configurable, I realized that `DatasetFactory` should be used only when talking to remote repositories, and `DatasetRepository` implementations should have independent low-level control over how they create dataset instances. Attempting to combine the two just results in configuration and dependency nightmare.